syz-manager/manager.go

```diff
@@ -991,10 +998,16 @@ func (mgr *Manager) collectSyscallInfoUnlocked() map[string]*CallCov {
 	return calls
 }
 
-func (mgr *Manager) fuzzerConnect() ([]rpctype.RPCInput, BugFrames) {
+func (mgr *Manager) fuzzerConnect(args *rpctype.ConnectArgs) ([]rpctype.RPCInput, BugFrames) {
 	mgr.mu.Lock()
 	defer mgr.mu.Unlock()
 
+	var idx int
+	_, err := fmt.Sscanf(args.Name, "vm-%d", &idx)
+	if err == nil {
+		mgr.machineInfo[idx] = args.MachineInfo
```

Comment:
We already have Fuzzer struct in rpc.go. I think it's better to store the machine info in that struct. Adding lots of parallel maps with single data item in rpc.go and manager.go won't scale well.

Manager.runInstance looks like a good place to communicate with RPCServer and reset/obtain machine info.
runInstance could generate name "vm-N"; then start the VM, call MonitorExecution, Close the VM; then ask  RPCServer for any state associated with this VM (currently will be only machine info). This will also be a good place to reset any per-VM state in RPCServer, so that we don't get a stale info on next run somehow.

Reply:
Got it. But just want to ask for "then ask RPCServer for any state associated with this VM", do you mean to store a reference to the server somewhere and call a server method, or to call the server via RPC? Because if it is the latter, I think I will also need to connect as a client from syz-manager because currently the server can call functions of the manager but not the other way back.

Reply:
Yes, just storing a reference to the rpc server and calling a method.
I assumed we already have that reference, but it seems we don't. It looks reasonable to add it.